src/basic/mountpoint-util.c

```diff
@@ -509,3 +512,19 @@ int mount_propagation_flags_from_string(const char *name, unsigned long *ret) {
                 return -EINVAL;
         return 0;
 }
+
+int make_mount_point_inode(const char *source, const char *dest, mode_t mode)
+{
+        struct stat st;
+
+        assert(source);
+        assert(dest);
+
+        if (stat(source, &st) < 0)
+                return -errno;
+
+        if (S_ISDIR(st.st_mode))
+                return mkdir_p_label(dest, mode);
```

Comment:
hmm, i think this should be a regular mkdir_label(), and the caller should call mkdir_parents() if it wants. I don't think we should create prefix dirs here unconditionally.